samples/matter/lock/src/app_task.cpp

```diff
@@ -404,61 +406,45 @@ void AppTask::StartBLEAdvertisingHandler()
 		return;
 	}
 
-	if (chip::Server::GetInstance().GetCommissioningWindowManager().OpenBasicCommissioningWindow() !=
-	    CHIP_NO_ERROR) {
+	if (Server::GetInstance().GetCommissioningWindowManager().OpenBasicCommissioningWindow() != CHIP_NO_ERROR) {
 		LOG_ERR("OpenBasicCommissioningWindow() failed");
 	}
 }
 
-void AppTask::LEDStateUpdateHandler(LEDWidget &ledWidget)
+void AppTask::UpdateLedStateEventHandler(const AppEvent &event)
 {
-	sAppTask.PostEvent(AppEvent{ AppEvent::UpdateLedState, &ledWidget });
+	if (event.Type == AppEventType::UpdateLedState) {
+		event.UpdateLedStateEvent.LedWidget->UpdateState();
+	}
 }
 
-void AppTask::LockStateChanged(BoltLockManager::State state, BoltLockManager::OperationSource source)
+void AppTask::LEDStateUpdateHandler(LEDWidget &ledWidget)
 {
-	switch (state) {
-	case BoltLockManager::State::kLockingInitiated:
-		LOG_INF("Lock action initiated");
-		sLockLED.Blink(50, 50);
-		break;
-	case BoltLockManager::State::kLockingCompleted:
-		LOG_INF("Lock action completed");
-		sLockLED.Set(true);
-		break;
-	case BoltLockManager::State::kUnlockingInitiated:
-		LOG_INF("Unlock action initiated");
-		sLockLED.Blink(50, 50);
-		break;
-	case BoltLockManager::State::kUnlockingCompleted:
-		LOG_INF("Unlock action completed");
-		sLockLED.Set(false);
-		break;
-	}
-
-	/* Handle changing attribute state in the application */
-	sAppTask.UpdateClusterState(state, source);
+	AppEvent event;
+	event.Type = AppEventType::UpdateLedState;
+	event.Handler = UpdateLedStateEventHandler;
+	event.UpdateLedStateEvent.LedWidget = &ledWidget;
+	PostEvent(event);
 }
 
 void AppTask::UpdateStatusLED()
 {
-#ifdef CONFIG_STATE_LEDS
 	/* Update the status LED.
 	 *
-	 * If thread and service provisioned, keep the LED On constantly.
+	 * If Thread and service provisioned, keep the LED On constantly.
```

Comment:
This sample doesn't apply to Thread only anymore. Maybe if IPv6 networking provisioned?